### ISSUE TITLE
Increase "number of uses" in command palette matcher

### ISSUE THREAD

**Issue by choldgraf** [jupyterlab/jupyterlab#4365]
This is related to #4362 , and perhaps is a more reasonable path to resolving my previous situation.

The main reason that the previous point came up is because I've found that I've started to use the command palette in the same way that I use Chrome's "awesome bar". E.g., hit `ctrl+command+c`, quickly type ~3 letters, and immediately hit "enter" without thinking about it.

This works great for creating a more rich version of keyboard shortcuts, but it only works if the same three-letter command always maps on to the same action, thus my original issue. I wonder if it'd be possible to upweight "number of times a command has been run with the command palette" so that the fuzzy matcher favors commands that have been run lots of times in recent history.

@sccolbert does something like this sound more reasonable, or technically possible w/o a large delta to the code?

**Comment by sccolbert** [comment]
That should be reasonable to implement. Can you move this issue to the `@phosphor/widgets` repo?

**Comment by jasongrout** [comment]
It would involve storing metrics about how much each command was used. I think it makes sense to have a "recent" list, but some thought should be given to how to store and persist that usage data.

**Comment by ian-r-rose** [comment]
I wonder if we can store some limited set of "favorite commands" in the SettingRegistry. It would not have to be very granular to be useful.

**Comment by jasongrout** [comment]
*and* you could then copy that to a new jlab deployment to get your history prepopulated.

**Comment by choldgraf** [comment]
@ian-r-rose yeah my guess is that most people would use at most a handful of commands most often

**Comment by sccolbert** [comment]
I can add a `frequency` field to the command item added to the palette, along with an option to initialize the frequency. This should be enough for you to persist the data on close.

**Comment by sccolbert** [comment]
Perhaps better is just adding the concept of execution count to a command itself...

**Comment by sccolbert** [comment]
Then using that metadata for rendering where applicable. (i.e. here)
